drivers/i2s/i2s_litex.c

```diff
@@ -0,0 +1,617 @@
+/*
+ * Copyright (c) 2020 Antmicro <www.antmicro.com>
+ * 
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <drivers/i2s.h>
+#include <sys/byteorder.h>
+#include <soc.h>
+#include <sys/util.h>
+#include <assert.h>
+#include "i2s_litex.h"
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(i2s_litex);
+#define DEV_CFG(dev) ((struct i2s_litex_cfg *const)(dev)->config_info)
+#define DEV_DATA(dev) ((struct i2s_litex_data *const)(dev)->driver_data)
+
+#define MODULO_INC(val, max)                                                   \
+	{                                                                      \
+		val = (++val < max) ? val : 0;                                 \
```

Comment:
optional: this is hard to read.  How about

val = (val == max - 1) ? 0 : val + 1

Reply:
Refactored as suggested